[ML] Migrate internal urls to non-hash paths

qn895 opened the pull request:
## Summary

This is WIP to migrate ML internal URLs to non-hash paths. Follow up of https://github.com/elastic/kibana/pull/75696 and part of meta issue to switch from HashRouter to BrowserRouter https://github.com/elastic/kibana/pull/75696.

- [x] Change the title of the pages (docTitle) whenever navigating to a new tab
  - [x] Only persist the `refreshInterval` in the `_g` parameter
- [x] Fix some issue with router history not navigating back correctly for some pages (e.g. Filters/Calendar pages)
- [x] Fix broken links
  - [x] Start trial link in data visualizer
  - [x] Select a different index link when creating indices that are not time based
- [x] Replace the links for both Anomaly and DFA tables. Should work correctly for:
  - [x] overview page
  - [x] AD/DFA job list pages
  - [x] Kibana management page
- [x] Links to jobs when creating or viewing from Kibana's sample data
- [x] Capabilities resolvers
  - [x] adds `redirectToMlAccessDeniedPage` as a new field in `PageDependencies`
  - [x] checkGetJobsCapabilitiesResolver -> adds `redirectToMlAccessDeniedPage` as required arg
  - [x] checkCreateJobsCapabilitiesResolver -> adds `redirectToMlAccessDeniedPage` as required arg
  - [x] checkFindFileStructurePrivilegeResolver -> adds `redirectToJobsManagementPage` as required arg
  - [x] checkMlNodesAvailable -> adds `redirectToJobsManagementPage` as required arg
- [x] Links from Data Visualizer results to 
  - [x] Create new job based on index
  - [x] View in data visualizer from file imports
  - [x] Create new job from file imports
- [x] Calendars & Filters redirects (manage, create, cancel, new, save)
- [x] Link to calendars from the Job Details expanded panel
- [x] Link to SMV from the Job Details - Forecast panel
- [x] Link to SMV from the Job Details - Annotations table
- [x] Link to SMV from the Anomaly Explorer - Anomalies chart
- [x] Switching between the SMV and Anomaly Explorer

### Potential Follow-up PRs
- [ ] Persist the refreshInterval setting even after creating a new AD job (currently it's being reset-ed to { pause: true, value: 0 }

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/)
* Commit: 8ffd123d52a386be610920357f7e7acdda27d3f1
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/execution/node/362/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/","number":"72506","commit":"8ffd123d52a386be610920357f7e7acdda27d3f1"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/)
* Commit: 25272ca22440e58dc52c0e09be0b1b88a41eb535
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/execution/node/358/log)

### History
* :broken_heart: [Build #72506](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/) failed 8ffd123d52a386be610920357f7e7acdda27d3f1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/","number":"72506","commit":"8ffd123d52a386be610920357f7e7acdda27d3f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/","number":"72775","commit":"25272ca22440e58dc52c0e09be0b1b88a41eb535"}]}
PIPELINE-->

walterra commented on x-pack/plugins/ml/common/constants/data_frame_analytics.ts:
```diff
@@ -0,0 +1,11 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const ANALYSIS_CONFIG_TYPE = {
+  OUTLIER_DETECTION: 'outlier_detection',
+  REGRESSION: 'regression',
+  CLASSIFICATION: 'classification',
+} as const;
```
We have the same as an `enum` already in `plugins/ml/public/application/data_frame_analytics/common/analytics.ts`. This one using the `const` approach as well as the location here LGTM so suggest to try to get rid of the on mentioned in this comment.

qn895 replied:
Went through and updated the usage of enum ANALYSIS_CONFIG_TYPE to const ANALYSIS_CONFIG_TYPE here https://github.com/elastic/kibana/pull/76735/commits/dd3517b44446099977761548fc7adf3f0fd4ab6c

walterra commented on x-pack/plugins/ml/common/types/data_frame_analytics.ts:
```diff
@@ -79,3 +80,5 @@ export interface DataFrameAnalyticsConfig {
   allow_lazy_start?: boolean;
 }
+
+export type DataFrameAnalyticsType = typeof ANALYSIS_CONFIG_TYPE[keyof typeof ANALYSIS_CONFIG_TYPE];
```
I don't think we use the `Type` postfix in other places so suggest to just name this `DataFrameAnalytics`.

walterra commented on x-pack/plugins/ml/public/application/components/navigation_menu/main_tabs.tsx:
```diff
@@ -87,16 +118,32 @@ export const MainTabs: FC<Props> = ({ tabId, disableLinks }) => {
+  };
+
+  useEffect(() => {
+    document.title = `ML - ${TAB_DATA[selectedTabId].name} - Kibana`;
```
I wonder if Kibana offers a way to do this more "officially" without directly manipulating `document.title`?

qn895 replied:
That's a good point! I do see `coreStart.docTitle.change('My Title')` that the Kibana team has exposed out ... Gonna take a look to see if we an implement this using that instead 👍

qn895 replied:
Changed to using `chrome.docTitle.change()` here https://github.com/elastic/kibana/pull/76735/commits/c62a924a86d160f7f5a8804fcf8d68285d3aef42

walterra commented on x-pack/plugins/ml/public/application/components/navigation_menu/main_tabs.tsx:
```diff
@@ -87,16 +118,32 @@ export const MainTabs: FC<Props> = ({ tabId, disableLinks }) => {
+        const defaultPathId = (TAB_DATA[id].pathId || id) as MlUrlGeneratorState['page'];
         // globalState (e.g. selected jobs and time range) should be retained when changing pages.
         // appState will not be considered.
-        const fullGlobalStateString = globalState !== undefined ? `?_g=${encode(globalState)}` : '';
```
Why is the global state no longer necessary?

qn895 replied:
I'm still retaining the globalState here, just within `redirectToTab` instead of the encoded string.

walterra reviewed (commented):
Had a first look, overall the refactor and use of utils looks good. I added some smaller questions/suggestions.

One thing i noticed I'm not sure why this is necessary: I see the use of `async/await` with `navigateToUrl` or `await mlUrlGenerator.createUrl` but I'm not sure this is required?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/)
* Commit: 4a50f1f831fc878587f603e21cc55ca60db7185a

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/4a50f1f831fc878587f603e21cc55ca60db7185a) | diff | [baseline](https://github.com/elastic/kibana/commit/caa4e0c11b430719cff97a42ecac5548c8d3cf4d) |
| --- | --- | --- | --- |
| ml | 8.2MB | +11.1KB | 8.2MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/4a50f1f831fc878587f603e21cc55ca60db7185a) | diff | [baseline](https://github.com/elastic/kibana/commit/caa4e0c11b430719cff97a42ecac5548c8d3cf4d) |
| --- | --- | --- | --- |
| ml | 608.6KB | +24.4KB | 584.2KB |

#### distributable file count
| id | [value](https://github.com/elastic/kibana/commit/4a50f1f831fc878587f603e21cc55ca60db7185a) | diff | [baseline](https://github.com/elastic/kibana/commit/caa4e0c11b430719cff97a42ecac5548c8d3cf4d) |
| --- | --- | --- | --- |
| default | 45455 | +1 | 45454 |

### History
* :broken_heart: [Build #72775](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/) failed 25272ca22440e58dc52c0e09be0b1b88a41eb535
* :broken_heart: [Build #72506](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/) failed 8ffd123d52a386be610920357f7e7acdda27d3f1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/","number":"72506","commit":"8ffd123d52a386be610920357f7e7acdda27d3f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/","number":"72775","commit":"25272ca22440e58dc52c0e09be0b1b88a41eb535"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/","number":"73466","commit":"4a50f1f831fc878587f603e21cc55ca60db7185a"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/)
* Commit: de7c2685f54d5035151512e8d4885d143129ec31
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/execution/node/352/log)

### History
* :green_heart: [Build #73466](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/) succeeded 4a50f1f831fc878587f603e21cc55ca60db7185a
* :broken_heart: [Build #72775](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/) failed 25272ca22440e58dc52c0e09be0b1b88a41eb535
* :broken_heart: [Build #72506](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/) failed 8ffd123d52a386be610920357f7e7acdda27d3f1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/","number":"72506","commit":"8ffd123d52a386be610920357f7e7acdda27d3f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/","number":"72775","commit":"25272ca22440e58dc52c0e09be0b1b88a41eb535"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/","number":"73466","commit":"4a50f1f831fc878587f603e21cc55ca60db7185a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/","number":"73546","commit":"de7c2685f54d5035151512e8d4885d143129ec31"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/)
* Commit: 853d5a3e1238f4c2a20b400df1144bf39866738f
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/execution/node/348/log)

### History
* :broken_heart: [Build #73546](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/) failed de7c2685f54d5035151512e8d4885d143129ec31
* :green_heart: [Build #73466](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/) succeeded 4a50f1f831fc878587f603e21cc55ca60db7185a
* :broken_heart: [Build #72775](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/) failed 25272ca22440e58dc52c0e09be0b1b88a41eb535
* :broken_heart: [Build #72506](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/) failed 8ffd123d52a386be610920357f7e7acdda27d3f1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/","number":"72506","commit":"8ffd123d52a386be610920357f7e7acdda27d3f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/","number":"72775","commit":"25272ca22440e58dc52c0e09be0b1b88a41eb535"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/","number":"73466","commit":"4a50f1f831fc878587f603e21cc55ca60db7185a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/","number":"73546","commit":"de7c2685f54d5035151512e8d4885d143129ec31"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/","number":"73568","commit":"853d5a3e1238f4c2a20b400df1144bf39866738f"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/)
* Commit: 662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/execution/node/342/log)

### History
* :broken_heart: [Build #73568](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/) failed 853d5a3e1238f4c2a20b400df1144bf39866738f
* :broken_heart: [Build #73546](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/) failed de7c2685f54d5035151512e8d4885d143129ec31
* :green_heart: [Build #73466](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/) succeeded 4a50f1f831fc878587f603e21cc55ca60db7185a
* :broken_heart: [Build #72775](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/) failed 25272ca22440e58dc52c0e09be0b1b88a41eb535
* :broken_heart: [Build #72506](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/) failed 8ffd123d52a386be610920357f7e7acdda27d3f1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72506/","number":"72506","commit":"8ffd123d52a386be610920357f7e7acdda27d3f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/","number":"72775","commit":"25272ca22440e58dc52c0e09be0b1b88a41eb535"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/","number":"73466","commit":"4a50f1f831fc878587f603e21cc55ca60db7185a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/","number":"73546","commit":"de7c2685f54d5035151512e8d4885d143129ec31"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/","number":"73568","commit":"853d5a3e1238f4c2a20b400df1144bf39866738f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/","number":"73578","commit":"662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7"}]}
PIPELINE-->

pheyos commented on x-pack/plugins/ml/public/application/settings/filter_lists/list/table.js:
```diff
@@ -213,7 +214,7 @@ export function FilterListsTable({
             data-test-subj="mlFilterListsTable"
             rowProps={(item) => ({
-              'data-test-subj': `mlCalendarListRow row-${item.filter_id}`,
+              'data-test-subj': `mlFilterListsTable row-${item.filter_id}`,
```
```suggestion
              'data-test-subj': `mlFilterListsRow row-${item.filter_id}`,
```
We already use `mlFilterListsTable` for the whole table. This test subject is for the row only.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/76735/commits/dc3ba76792563da7c40db99ac11c6d69100f3bd7

peteharverson commented on x-pack/plugins/ml/public/application/routing/routes/data_frame_analytics/analytics_job_exploration.tsx:
```diff
@@ -38,16 +39,31 @@ const PageWrapper: FC<PageProps> = ({ location, deps }) => {
-    console.error('Could not parse global state');
-    window.location.href = '#data_frame_analytics';
+    console.error(
+      'Could not parse global state. Redirecting to Dataframe Analytics Management Page.'
```
Nit as this is only shown in the console, but we use two words for `Data Frame`.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/76735/commits/dc3ba76792563da7c40db99ac11c6d69100f3bd7

peteharverson commented on x-pack/plugins/ml/public/application/components/navigation_menu/main_tabs.tsx:
```diff
@@ -66,59 +69,116 @@ function getTabs(disableLinks: boolean): Tab[] {
+  };
+
+  useEffect(() => {
+    docTitle.change([TAB_DATA[selectedTabId].name, ML_APP_NAME, 'Elastic']);
```
I am seeing a duplicate 'Elastic' for all my tab titles:

![image](https://user-images.githubusercontent.com/7405507/92764503-63968000-f38c-11ea-91d7-ec922f8f07c3.png)

qn895 replied:
Fixed here https://github.com/elastic/kibana/pull/76735/commits/dc3ba76792563da7c40db99ac11c6d69100f3bd7

qn895 commented:
> The 'Recently viewed' link for the SMV here has an extra `ml#/timeseriesexplorer/` portion. Do we have control over that URL?

This is now updated here https://github.com/elastic/kibana/pull/76735/commits/dc3ba76792563da7c40db99ac11c6d69100f3bd7! Thanks - I missed that. However clicking the url now takes a bit long to load and is not instant. I suspect we can investigate this issue further in a follow up PR.

I also added the fix for broken breadcrumb and navigation issues mentioned in https://github.com/elastic/kibana/issues/72013 and introduced another level to the navigation for Calendar Management/Filter List here https://github.com/elastic/kibana/pull/76735/commits/dc3ba76792563da7c40db99ac11c6d69100f3bd7

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/)
* Commit: 40f884594e1d4023a8aacd5cca779e789517bda2
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/execution/node/368/log)

### History
* :broken_heart: [Build #73578](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/) failed 662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7
* :broken_heart: [Build #73568](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/) failed 853d5a3e1238f4c2a20b400df1144bf39866738f
* :broken_heart: [Build #73546](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/) failed de7c2685f54d5035151512e8d4885d143129ec31
* :green_heart: [Build #73466](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/) succeeded 4a50f1f831fc878587f603e21cc55ca60db7185a
* :broken_heart: [Build #72775](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/) failed 25272ca22440e58dc52c0e09be0b1b88a41eb535

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72775/","number":"72775","commit":"25272ca22440e58dc52c0e09be0b1b88a41eb535"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/","number":"73466","commit":"4a50f1f831fc878587f603e21cc55ca60db7185a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/","number":"73546","commit":"de7c2685f54d5035151512e8d4885d143129ec31"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/","number":"73568","commit":"853d5a3e1238f4c2a20b400df1144bf39866738f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/","number":"73578","commit":"662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/","number":"73999","commit":"40f884594e1d4023a8aacd5cca779e789517bda2"}]}
PIPELINE-->

pheyos commented on x-pack/test/functional/services/ml/settings_filter_list.ts:
```diff
@@ -17,7 +17,7 @@ export function MachineLearningSettingsFilterListProvider({ getService }: FtrPro
       const rows = [];
 
-      for (const tr of $.findTestSubjects('~mlCalendarListRow').toArray()) {
+      for (const tr of $.findTestSubjects('~mlFilterListsTable').toArray()) {
```
This should also be updated to `mlFilterListsRow`.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/76735/commits/1fa702690b0cd8a4a4b65498cb2fad5d6d46f5df

darnautov commented on x-pack/plugins/ml/common/types/ml_url_generator.ts:
```diff
@@ -5,27 +5,21 @@
 
 export type MLPageState<PageType, PageState> = PageState extends OptionalPageState
-  ? { page: PageType; pageState?: PageState }
+  ? { page: PageType; pageState?: PageState; excludeBasePath?: boolean }
   : PageState extends object
-  ? { page: PageType; pageState: PageState }
+  ? { page: PageType; pageState: PageState; excludeBasePath?: boolean }
   : { page: PageType };
```
I guess `excludeBasePath` is applicable for any page, isn't it?

qn895 replied:
You're right. Updated here https://github.com/elastic/kibana/pull/76735/commits/cea397ee78ebf873893a6bc05185ddacaa40704c

darnautov commented on x-pack/plugins/ml/public/application/app.tsx:
```diff
@@ -50,11 +50,17 @@ export interface MlServicesContext {
 
 const App: FC<AppProps> = ({ coreStart, deps, appMountParams }) => {
+  const redirectToMlAccessDeniedPage = async () => {
+    const accessDeniedPageUrl = `${coreStart.http.basePath.get()}/app/ml/access-denied`;
```
I think this URL should be provided by the ML URL generator

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/76735/commits/cea397ee78ebf873893a6bc05185ddacaa40704c

darnautov commented on x-pack/plugins/ml/public/application/capabilities/check_capabilities.ts:
```diff
@@ -33,10 +33,12 @@ export function checkGetManagementMlJobsResolver() {
+export function checkGetJobsCapabilitiesResolver(
+  redirectToMlAccessDeniedPage: () => Promise<void>
+): Promise<MlCapabilities> {
   return new Promise((resolve, reject) => {
     getCapabilities()
-      .then(({ capabilities, isPlatinumOrTrialLicense }) => {
+      .then(async ({ capabilities, isPlatinumOrTrialLicense }) => {
```
There is no need to wrap in Promise, just an async function with `try...catch`

darnautov commented on x-pack/plugins/ml/public/application/components/data_recognizer/recognized_result.js:
```diff
@@ -9,11 +9,16 @@ import PropTypes from 'prop-types';
   const id = savedSearch === null ? `index=${indexPattern.id}` : `savedSearchId=${savedSearch.id}`;
-
-  const href = `#/jobs/new_job/recognize?id=${config.id}&${id}`;
+  const href = `${basePath.get()}/app/ml/jobs/new_job/recognize?id=${config.id}&${id}`;
```
this URL should be created by the ML URL generator

darnautov commented on x-pack/plugins/ml/public/application/components/navigation_menu/main_tabs.tsx:
```diff
@@ -25,6 +24,10 @@ interface Props {
   tabId: TabId;
 }
 
+const ML_APP_NAME = i18n.translate('xpack.ml.navMenu.mlAppNameText', {
```
should be placed in the constants folder

qn895 replied:
Moved the constants here https://github.com/elastic/kibana/pull/76735/commits/68e75199438fe6ce84017c80903ad2eaa43cfdc7

darnautov commented on x-pack/plugins/ml/public/application/components/navigation_menu/main_tabs.tsx:
```diff
@@ -66,59 +69,116 @@ function getTabs(disableLinks: boolean): Tab[] {
+            },
+          }
+        : undefined;
+    // @ts-ignore
```
please avoid using `@ts-ignore` here

darnautov commented on x-pack/plugins/ml/public/application/components/rule_editor/scope_section.js:
```diff
@@ -17,8 +17,19 @@ import { ScopeExpression } from './scope_expression';
+  const navigateToPath = useNavigateToPath();
+  const redirectToFilterManagementPage = async () => {
+    const path = await mlUrlGenerator.createUrl({
+      page: ML_PAGES.FILTER_LISTS_MANAGE,
```
nit: I suppose it'd be more convenient to have `ML_PAGES` as a property of `mlUrlGenerator` to avoid extra static import

qn895 replied:
I think I didn't do this here because of the limitations with  `UrlGeneratorContract<"ML_APP_URL_GENERATOR">`. Let me reach out to Anton and see if we can do something to our contract so others can access it additional properties.

darnautov commented on x-pack/plugins/ml/public/application/datavisualizer/file_based/components/results_links/results_links.tsx:
```diff
@@ -37,11 +39,37 @@ export const ResultsLinks: FC<Props> = ({
+    await navigateToPath(path);
+  }, [indexPatternId, globalState]);
+
+  const createNewMlJob = useCallback(async () => {
```
I find this name misleading, it doesn't actually create a new ML job

qn895 replied:
Renamed here https://github.com/elastic/kibana/pull/76735/commits/cea397ee78ebf873893a6bc05185ddacaa40704c

darnautov commented on x-pack/plugins/ml/public/application/datavisualizer/file_based/components/results_links/results_links.tsx:
```diff
@@ -89,7 +124,7 @@ export const ResultsLinks: FC<Props> = ({
             }
             description=""
-            href={`${basePath.get()}/app/discover#/?&_a=(index:'${indexPatternId}')${globalStateString}`}
+            href={`${basePath.get()}/app/discover/?&_a=(index:'${indexPatternId}')${globalStateString}`}
```
please use the URL generator provided by the Discover plugin

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/76735/commits/1efaed29b3be3a6a6a9fa6ae27caf30815bc3070

darnautov commented on x-pack/plugins/ml/public/application/explorer/components/explorer_no_jobs_found/explorer_no_jobs_found.js:
```diff
@@ -22,9 +23,14 @@ export const ExplorerNoJobsFound = () => (
-      <EuiButton color="primary" fill href="ml#/jobs">
-        <FormattedMessage id="xpack.ml.explorer.createNewJobLinkText" defaultMessage="Create job" />
-      </EuiButton>
+      <Link to={`/${ML_PAGES.ANOMALY_DETECTION_JOBS_MANAGE}`}>
```
I think we should use a link from the URL generator

darnautov commented on x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_actions/results.js:
```diff
@@ -37,29 +44,59 @@ export function ResultLinks({ jobs }) {
         <EuiToolTip position="bottom" content={openJobsInSingleMetricViewerText}>
+          {isManagementTable ? (
+            <EuiButtonIcon
+              href={`${basePath.get()}/app/ml/${timeSeriesExplorerLink}`}
```
Link should be created by ML URL generator

qn895 replied:
I'll keep this here for now because these are being generated using `mlJobService.createResultsUrlForJobs` and we need both versions (with and without `basePath`) so even if we use the MlUrlGenerator, we will have to call this line.

darnautov replied:
I believe it will make code easier to maintain having a single source of truth for URLs. We can inject `mlJobService` into the URL generator and invoke `createResultsUrlForJobs` from there.

darnautov commented on x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_actions/results.js:
```diff
@@ -37,29 +44,59 @@ export function ResultLinks({ jobs }) {
           <EuiButtonIcon
-            href={createLinkWithUserDefaults('timeseriesexplorer', jobs)}
-            iconType="visLine"
+            href={`${basePath.get()}/app/ml/${anomalyExplorerLink}`}
```
Link should be created by ML URL generator

qn895 replied:
Same as above

qn895 commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/use_columns.tsx:
```diff
@@ -134,9 +134,14 @@ export const progressColumn = {
+    pageState: { jobId: item.id },
+  });
+
+  return <EuiLink href={href}>{item.id}</EuiLink>;
```
After discussing with Pete, I think we want to keep the behavior as it is now for the current PR.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/)
* Commit: 0c29605a135bb1f1ab050afe7f95505e675c9813
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/execution/node/368/log)

### History
* :broken_heart: [Build #73999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/) failed 40f884594e1d4023a8aacd5cca779e789517bda2
* :broken_heart: [Build #73578](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/) failed 662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7
* :broken_heart: [Build #73568](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/) failed 853d5a3e1238f4c2a20b400df1144bf39866738f
* :broken_heart: [Build #73546](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/) failed de7c2685f54d5035151512e8d4885d143129ec31
* :green_heart: [Build #73466](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/) succeeded 4a50f1f831fc878587f603e21cc55ca60db7185a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73466/","number":"73466","commit":"4a50f1f831fc878587f603e21cc55ca60db7185a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/","number":"73546","commit":"de7c2685f54d5035151512e8d4885d143129ec31"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/","number":"73568","commit":"853d5a3e1238f4c2a20b400df1144bf39866738f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/","number":"73578","commit":"662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/","number":"73999","commit":"40f884594e1d4023a8aacd5cca779e789517bda2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/","number":"74275","commit":"0c29605a135bb1f1ab050afe7f95505e675c9813"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/)
* Commit: f85413f043c88047b62176c5025f9803679547bd
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/execution/node/368/log)

### History
* :broken_heart: [Build #74275](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/) failed 0c29605a135bb1f1ab050afe7f95505e675c9813
* :broken_heart: [Build #73999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/) failed 40f884594e1d4023a8aacd5cca779e789517bda2
* :broken_heart: [Build #73578](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/) failed 662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7
* :broken_heart: [Build #73568](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/) failed 853d5a3e1238f4c2a20b400df1144bf39866738f
* :broken_heart: [Build #73546](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/) failed de7c2685f54d5035151512e8d4885d143129ec31

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73546/","number":"73546","commit":"de7c2685f54d5035151512e8d4885d143129ec31"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/","number":"73568","commit":"853d5a3e1238f4c2a20b400df1144bf39866738f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/","number":"73578","commit":"662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/","number":"73999","commit":"40f884594e1d4023a8aacd5cca779e789517bda2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/","number":"74275","commit":"0c29605a135bb1f1ab050afe7f95505e675c9813"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/","number":"74312","commit":"f85413f043c88047b62176c5025f9803679547bd"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/)
* Commit: 1efaed29b3be3a6a6a9fa6ae27caf30815bc3070

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/1efaed29b3be3a6a6a9fa6ae27caf30815bc3070) | diff | [baseline](https://github.com/elastic/kibana/commit/1831fb3d2dadae43350a1ac77d085760a762120d) |
| --- | --- | --- | --- |
| ml | 8.0MB | +32.2KB | 8.0MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/1efaed29b3be3a6a6a9fa6ae27caf30815bc3070) | diff | [baseline](https://github.com/elastic/kibana/commit/1831fb3d2dadae43350a1ac77d085760a762120d) |
| --- | --- | --- | --- |
| ml | 872.0KB | +26.7KB | 845.3KB |

### History
* :broken_heart: [Build #74312](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/) failed f85413f043c88047b62176c5025f9803679547bd
* :broken_heart: [Build #74275](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/) failed 0c29605a135bb1f1ab050afe7f95505e675c9813
* :broken_heart: [Build #73999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/) failed 40f884594e1d4023a8aacd5cca779e789517bda2
* :broken_heart: [Build #73578](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/) failed 662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7
* :broken_heart: [Build #73568](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/) failed 853d5a3e1238f4c2a20b400df1144bf39866738f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73568/","number":"73568","commit":"853d5a3e1238f4c2a20b400df1144bf39866738f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/","number":"73578","commit":"662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/","number":"73999","commit":"40f884594e1d4023a8aacd5cca779e789517bda2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/","number":"74275","commit":"0c29605a135bb1f1ab050afe7f95505e675c9813"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/","number":"74312","commit":"f85413f043c88047b62176c5025f9803679547bd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/","number":"74320","commit":"1efaed29b3be3a6a6a9fa6ae27caf30815bc3070"}]}
PIPELINE-->

walterra commented on x-pack/plugins/ml/common/types/ml_url_generator.ts:
```diff
@@ -12,10 +12,10 @@ import { DataFrameAnalyticsType } from './data_frame_analytics';
-  ? { page: PageType; pageState: PageState }
-  : { page: PageType };
+  ? { page: PageType; pageState: PageState; excludeBasePath?: boolean }
+  : { page: PageType; excludeBasePath?: boolean };
```
We should try to avoid nested ternary operators (https://github.com/elastic/kibana/blob/7.9/STYLEGUIDE.md#only-use-ternary-operators-for-small-simple-code)

qn895 replied:
As of now I'm not sure what's a good replacement for this. I experimented with a few ways but this way as of right now is the most explicit type that covers all the requirements.

walterra commented on x-pack/plugins/ml/public/application/contexts/kibana/use_create_url.ts:
```diff
@@ -40,17 +48,32 @@ export const useCreateAndNavigateToMlLink = (
+            }
+          : undefined;
+
+      // @ts-ignore
```
If we cannot get rid of this `ignore` in this PR, can you please add a `// TODO` comment and describe the reason TS is failing?

qn895 replied:
Added here https://github.com/elastic/kibana/pull/76735/commits/0d83eac644d202155da42ecda4b39145a7a4fc0b

qn895 commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/models_management/models_list.tsx:
```diff
@@ -278,12 +280,17 @@ export const ModelsList: FC = () => {
-            Object.keys(item.metadata?.analytics_config.analysis)[0]
-          )
-        );
+        const url = await urlGenerator.createUrl({
```
Updated here https://github.com/elastic/kibana/pull/76735/commits/cea397ee78ebf873893a6bc05185ddacaa40704c

qn895 commented on x-pack/plugins/ml/public/application/contexts/kibana/use_create_url.ts:
```diff
@@ -18,3 +21,49 @@ export const useMlUrlGenerator = () => {
+  };
+
+  // returns the onClick callback
+  return useCallback(() => redirectToMlPage(page), [mlUrlGenerator, page]);
```
Updated here https://github.com/elastic/kibana/pull/76735/commits/cea397ee78ebf873893a6bc05185ddacaa40704c

walterra commented on x-pack/plugins/ml/public/application/app.tsx:
```diff
@@ -50,11 +51,21 @@ export interface MlServicesContext {
     indexPatterns: deps.data.indexPatterns,
     config: coreStart.uiSettings!,
     setBreadcrumbs: coreStart.chrome!.setBreadcrumbs,
+    redirectToMlAccessDeniedPage,
```
We now have two different approaches how this is accessible, one time it's used via hooks (`const redirectToMlAccessDeniedPage = useCreateAndNavigateToMlLink(ML_PAGES.ACCESS_DENIED);`) and the other approach is defined here via `deps`. I wonder if we have it available in `deps` anyway if we shouldn't use it via that all the time?

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/76735/commits/f762277c076b96176a2b7159cf577d6a6854b870

qn895 commented on x-pack/plugins/ml/public/application/contexts/kibana/use_create_url.ts:
```diff
@@ -18,3 +21,49 @@ export const useMlUrlGenerator = () => {
+    setHref(url);
+  }, []);
+  useEffect(() => {
+    generateUrl();
```
Added a `isCancelled` flag to the effect here so it won't setState if the component is unmounted. Not sure if this is the best solution: https://github.com/elastic/kibana/pull/76735/commits/cea397ee78ebf873893a6bc05185ddacaa40704c

qn895 replied:
Revised some more here per our conversation: https://github.com/elastic/kibana/pull/76735/commits/42313aed218ac386fc52710b8bf95592744abd56

peteharverson commented on x-pack/plugins/ml/public/application/util/recently_accessed.ts:
```diff
@@ -37,7 +37,7 @@ export function addItemToRecentlyAccessed(page: string, itemId: string, url: str
   }
 
-  url = `ml#/${page}/${url}`;
+  url = `${page}/${url}`;
```
This fails if clicking this link from outside the ML app. E.g. go back to the Kibana home page, and click on an ML recently accessed link, the `ml` is now needed in the url:

![image](https://user-images.githubusercontent.com/7405507/93229252-4d166d00-f76e-11ea-977a-bf035b80a557.png)

qn895 replied:
Thanks for catching that. I didn't test this outside of the ML plugin so the path was different. Fixed the issue here https://github.com/elastic/kibana/pull/76735/commits/ba56a81be63cd5517d3dd4be1fc9d2409b26b351

walterra commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/models_management/models_list.tsx:
```diff
@@ -278,12 +280,19 @@ export const ModelsList: FC = () => {
+            jobId: item.metadata?.analytics_config.id as string,
+            analysisType: Object.keys(
+              item.metadata?.analytics_config.analysis
+            )[0] as DataFrameAnalyticsType,
```
We have `getAnalysisType()` available in `plugins/ml/public/application/data_frame_analytics/common/analytics.ts`.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/76735/commits/cc8f5dd3c6567646ceea67439dd8544e8fb7789d

walterra commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/state.ts:
```diff
@@ -290,7 +291,7 @@ export function getFormStateFromJobConfig(
   isClone: boolean = true
 ): Partial<State['form']> {
-  const jobType = Object.keys(analyticsJobConfig.analysis)[0] as ANALYSIS_CONFIG_TYPE;
+  const jobType = Object.keys(analyticsJobConfig.analysis)[0] as DataFrameAnalyticsType;
```
We have `getAnalysisType()` available in `plugins/ml/public/application/data_frame_analytics/common/analytics.ts`.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/76735/commits/cc8f5dd3c6567646ceea67439dd8544e8fb7789d

walterra commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/services/analytics_service/get_analytics.ts:
```diff
@@ -143,7 +140,7 @@ export const getAnalyticsFactory = (
               config,
               id: config.id,
-              job_type: Object.keys(config.analysis)[0] as ANALYSIS_CONFIG_TYPE,
+              job_type: Object.keys(config.analysis)[0] as DataFrameAnalyticsType,
```
We have `getAnalysisType()` available in `plugins/ml/public/application/data_frame_analytics/common/analytics.ts`.

qn895 replied:
Updated here cc8f5dd

walterra commented on x-pack/plugins/ml/public/application/datavisualizer/file_based/components/results_links/results_links.tsx:
```diff
@@ -49,6 +53,38 @@ export const ResultsLinks: FC<Props> = ({
+  } = useMlKibana();
+
+  useEffect(() => {
+    let unamounted = false;
```
should this be `unmounted`?

qn895 replied:
Updated here cc8f5dd

walterra commented on x-pack/plugins/ml/public/application/explorer/components/explorer_no_jobs_found/explorer_no_jobs_found.js:
```diff
@@ -12,20 +12,36 @@ import React from 'react';
-  />
-);
+import { ML_PAGES } from '../../../../../common/constants/ml_url_generator';
+import { Link } from 'react-router-dom';
```
Nit: This import could be at the top of the file.

qn895 replied:
Updated here cc8f5dd

walterra commented on x-pack/plugins/ml/public/application/jobs/jobs_list/components/jobs_list/jobs_list.js:
```diff
@@ -71,7 +71,7 @@ export class JobsList extends Component {
     }
 
-    return <ADJobIdLink id={id} />;
+    return <ADJobIdLink key={id} id={id} />;
```
Nit: Besides `Ml` as a prefix we hardly use any abbreviations in component names so I was wondering if we should make this the full written `<AnomalyDetectionJobIdLink ... />`.

qn895 replied:
Updated here cc8f5dd

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/)
* Commit: d3edff7875b36ec9871d42c9ac12e9b489e64694

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/d3edff7875b36ec9871d42c9ac12e9b489e64694) | diff | [baseline](https://github.com/elastic/kibana/commit/0574efaa8d75d31bb29b4241c55e17128367702f) |
| --- | --- | --- | --- |
| ml | 8.0MB | +32.2KB | 8.0MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/d3edff7875b36ec9871d42c9ac12e9b489e64694) | diff | [baseline](https://github.com/elastic/kibana/commit/0574efaa8d75d31bb29b4241c55e17128367702f) |
| --- | --- | --- | --- |
| ml | 872.0KB | +26.7KB | 845.3KB |

### History
* :green_heart: [Build #74320](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/) succeeded 1efaed29b3be3a6a6a9fa6ae27caf30815bc3070
* :broken_heart: [Build #74312](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/) failed f85413f043c88047b62176c5025f9803679547bd
* :broken_heart: [Build #74275](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/) failed 0c29605a135bb1f1ab050afe7f95505e675c9813
* :broken_heart: [Build #73999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/) failed 40f884594e1d4023a8aacd5cca779e789517bda2
* :broken_heart: [Build #73578](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/) failed 662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73578/","number":"73578","commit":"662c75688ee0f1d2ecf5ec22d6b7497b2b9229d7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/","number":"73999","commit":"40f884594e1d4023a8aacd5cca779e789517bda2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/","number":"74275","commit":"0c29605a135bb1f1ab050afe7f95505e675c9813"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/","number":"74312","commit":"f85413f043c88047b62176c5025f9803679547bd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/","number":"74320","commit":"1efaed29b3be3a6a6a9fa6ae27caf30815bc3070"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/","number":"74479","commit":"d3edff7875b36ec9871d42c9ac12e9b489e64694"}]}
PIPELINE-->

peteharverson commented:
Gave latest edits a test, and all looks good. Only issue I have found is when coming into the Anomaly Detection jobs list from the Management app jobs list, and deleting the search query, the address bar URL ends up as e.g. 

```
http://localhost:5601/zvl/app/ml/ml#/jobs?_g=(refreshInterval:(pause:!f,value:30000))
```

The page loads fine, but you get the extra `ml#` in the URL.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74514/)
* Commit: 0d83eac644d202155da42ecda4b39145a7a4fc0b

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/0d83eac644d202155da42ecda4b39145a7a4fc0b) | diff | [baseline](https://github.com/elastic/kibana/commit/0574efaa8d75d31bb29b4241c55e17128367702f) |
| --- | --- | --- | --- |
| ml | 8.0MB | +31.6KB | 8.0MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/0d83eac644d202155da42ecda4b39145a7a4fc0b) | diff | [baseline](https://github.com/elastic/kibana/commit/0574efaa8d75d31bb29b4241c55e17128367702f) |
| --- | --- | --- | --- |
| ml | 871.9KB | +26.6KB | 845.3KB |

### History
* :green_heart: [Build #74479](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/) succeeded d3edff7875b36ec9871d42c9ac12e9b489e64694
* :green_heart: [Build #74320](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/) succeeded 1efaed29b3be3a6a6a9fa6ae27caf30815bc3070
* :broken_heart: [Build #74312](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/) failed f85413f043c88047b62176c5025f9803679547bd
* :broken_heart: [Build #74275](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/) failed 0c29605a135bb1f1ab050afe7f95505e675c9813
* :broken_heart: [Build #73999](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/) failed 40f884594e1d4023a8aacd5cca779e789517bda2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/73999/","number":"73999","commit":"40f884594e1d4023a8aacd5cca779e789517bda2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/","number":"74275","commit":"0c29605a135bb1f1ab050afe7f95505e675c9813"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/","number":"74312","commit":"f85413f043c88047b62176c5025f9803679547bd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/","number":"74320","commit":"1efaed29b3be3a6a6a9fa6ae27caf30815bc3070"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/","number":"74479","commit":"d3edff7875b36ec9871d42c9ac12e9b489e64694"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74514/","number":"74514","commit":"0d83eac644d202155da42ecda4b39145a7a4fc0b"}]}
PIPELINE-->

pheyos reviewed (approved):
Functional test fix LGTM

qn895 commented:
> Gave latest edits a test, and all looks good. Only issue I have found is when coming into the Anomaly Detection jobs list from the Management app jobs list, and deleting the search query, the address bar URL ends up as e.g.
> 
> ```
> http://localhost:5601/zvl/app/ml/ml#/jobs?_g=(refreshInterval:(pause:!f,value:30000))
> ```
> 
> The page loads fine, but you get the extra `ml#` in the URL.

Thanks for catching that. This issue should now be fixed in https://github.com/elastic/kibana/pull/76735/commits/92dad646a46abe1e422adaeaca793a0e25956550.

I also realized I missed a few spots in the process of working on this issue and have made updates to those:
- [x] Link to calendars from the Job Details expanded panel
- [x] Link to SMV from the Job Details - Forecast panel
- [x] Link to SMV from the Job Details - Annotations table
- [x] Link to SMV from the Anomaly Explorer - Anomalies chart
- [x] Switching between the SMV and Anomaly Explorer

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74978/)
* Commit: 0a0436b765269ceba1a4f070c3ce73e3f13079c0
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74978/execution/node/105/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74978/execution/node/111/log)

### History
* :green_heart: [Build #74514](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74514/) succeeded 0d83eac644d202155da42ecda4b39145a7a4fc0b
* :green_heart: [Build #74479](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/) succeeded d3edff7875b36ec9871d42c9ac12e9b489e64694
* :green_heart: [Build #74320](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/) succeeded 1efaed29b3be3a6a6a9fa6ae27caf30815bc3070
* :broken_heart: [Build #74312](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/) failed f85413f043c88047b62176c5025f9803679547bd
* :broken_heart: [Build #74275](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/) failed 0c29605a135bb1f1ab050afe7f95505e675c9813

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74275/","number":"74275","commit":"0c29605a135bb1f1ab050afe7f95505e675c9813"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/","number":"74312","commit":"f85413f043c88047b62176c5025f9803679547bd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/","number":"74320","commit":"1efaed29b3be3a6a6a9fa6ae27caf30815bc3070"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/","number":"74479","commit":"d3edff7875b36ec9871d42c9ac12e9b489e64694"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74514/","number":"74514","commit":"0d83eac644d202155da42ecda4b39145a7a4fc0b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74978/","number":"74978","commit":"0a0436b765269ceba1a4f070c3ce73e3f13079c0"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75018/)
* Commit: ee507b441a68b8201fb7725ec8989d915c572b35

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/ee507b441a68b8201fb7725ec8989d915c572b35) | diff | [baseline](https://github.com/elastic/kibana/commit/37527f94f343fdd48c0a4843f9a95325924a7040) |
| --- | --- | --- | --- |
| ml | 8.1MB | +54.9KB | 8.0MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/ee507b441a68b8201fb7725ec8989d915c572b35) | diff | [baseline](https://github.com/elastic/kibana/commit/37527f94f343fdd48c0a4843f9a95325924a7040) |
| --- | --- | --- | --- |
| ml | 872.3KB | +27.0KB | 845.3KB |

### History
* :broken_heart: [Build #74978](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74978/) failed 0a0436b765269ceba1a4f070c3ce73e3f13079c0
* :green_heart: [Build #74514](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74514/) succeeded 0d83eac644d202155da42ecda4b39145a7a4fc0b
* :green_heart: [Build #74479](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/) succeeded d3edff7875b36ec9871d42c9ac12e9b489e64694
* :green_heart: [Build #74320](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/) succeeded 1efaed29b3be3a6a6a9fa6ae27caf30815bc3070
* :broken_heart: [Build #74312](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/) failed f85413f043c88047b62176c5025f9803679547bd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74312/","number":"74312","commit":"f85413f043c88047b62176c5025f9803679547bd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/","number":"74320","commit":"1efaed29b3be3a6a6a9fa6ae27caf30815bc3070"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/","number":"74479","commit":"d3edff7875b36ec9871d42c9ac12e9b489e64694"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74514/","number":"74514","commit":"0d83eac644d202155da42ecda4b39145a7a4fc0b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74978/","number":"74978","commit":"0a0436b765269ceba1a4f070c3ce73e3f13079c0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75018/","number":"75018","commit":"ee507b441a68b8201fb7725ec8989d915c572b35"}]}
PIPELINE-->

walterra reviewed (approved):
Latest changes LGTM

peteharverson reviewed (approved):
Tested latest edits and LGTM

qn895 commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75253/)
* Commit: a39face9bc6c169dcbc1f3685fc4e4ddc7ea88a6

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/a39face9bc6c169dcbc1f3685fc4e4ddc7ea88a6) | diff | [baseline](https://github.com/elastic/kibana/commit/4bf0932500e1d74e42bb8e4e42ec61f53fbbfd4a) |
| --- | --- | --- | --- |
| ml | 8.1MB | +54.9KB | 8.0MB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/a39face9bc6c169dcbc1f3685fc4e4ddc7ea88a6) | diff | [baseline](https://github.com/elastic/kibana/commit/4bf0932500e1d74e42bb8e4e42ec61f53fbbfd4a) |
| --- | --- | --- | --- |
| ml | 872.3KB | +27.0KB | 845.3KB |

### History
* :green_heart: [Build #75018](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75018/) succeeded ee507b441a68b8201fb7725ec8989d915c572b35
* :broken_heart: [Build #74978](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74978/) failed 0a0436b765269ceba1a4f070c3ce73e3f13079c0
* :green_heart: [Build #74514](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74514/) succeeded 0d83eac644d202155da42ecda4b39145a7a4fc0b
* :green_heart: [Build #74479](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/) succeeded d3edff7875b36ec9871d42c9ac12e9b489e64694
* :green_heart: [Build #74320](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/) succeeded 1efaed29b3be3a6a6a9fa6ae27caf30815bc3070

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74320/","number":"74320","commit":"1efaed29b3be3a6a6a9fa6ae27caf30815bc3070"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74479/","number":"74479","commit":"d3edff7875b36ec9871d42c9ac12e9b489e64694"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74514/","number":"74514","commit":"0d83eac644d202155da42ecda4b39145a7a4fc0b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74978/","number":"74978","commit":"0a0436b765269ceba1a4f070c3ce73e3f13079c0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75018/","number":"75018","commit":"ee507b441a68b8201fb7725ec8989d915c572b35"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75253/","number":"75253","commit":"a39face9bc6c169dcbc1f3685fc4e4ddc7ea88a6"}]}
PIPELINE-->
